include/sample_rdd.h

```diff
@@ -0,0 +1,92 @@
+#ifndef CPARK_SAMPLE_RDD_H
+#define CPARK_SAMPLE_RDD_H
+
+#include <random>
+#include <vector>
+#include "base_rdd.h"
+#include "filter_rdd.h"
+#include "utils.h"
+namespace cpark {
+
+/**
+ * An Rdd holding the sampled data from an old rdd by a specific sample fraction
+ * @tparam R Type of the old Rdd.
+ * The type of the old Rdd `R`'s elements should be able to invoke random boolean function,
+ * the invoke result must be in boolean, and the element type shouldn't change after sampling.
+ */
+template <concepts::Rdd R>
+requires std::invocable<std::function<bool(int)>, utils::RddElementType<R>>&& std::is_convertible_v<
```

Comment:
This requirement would constrain the element type of the previous RDD to be merely `int`. I would suggest removing this requires clause. Please also see the comment in line 55.